src/fsharp/tast.fs

```diff
@@ -2907,10 +2901,10 @@ and [<NoEquality; NoComparison; StructuredFormatDisplay("{DebugText}")>]
         //         let dt = System.DateTime.Now - System.DateTime.Now // IsMemberOrModuleBinding = false, IsCompiledAsTopLevel = true, IsMember = false, CompilerGenerated=true
         //    
         // However we don't need this for CompilerGenerated members such as the implementations of IComparable
-        if x.IsCompiledAsTopLevel && not x.IsMember && (x.IsCompilerGenerated || not x.IsMemberOrModuleBinding) then 
-            globalStableNameGenerator.GetUniqueCompilerGeneratedName(givenName, x.Range, x.Stamp) 
-        else 
-            givenName
+        match x.val_global_state with
```

Comment:
I guess we should just pass `g` into `CompiledName`

Reply:
I would love to have passed around TcGlobals, however, it uses types from tast.fs, and so I can't pass tcGlobals that far back